canutils/slcan/slcan.c

```diff
@@ -316,9 +316,22 @@ int main(int argc, char *argv[])
                         {
                           /* open CAN interface */
 
-                          mode = 1;
-                          debug_print("Open interface\n");
-                          ok_return(fd);
+                          struct ifreq ifr;
+
+                          strlcpy(ifr.ifr_name, argv[1], IFNAMSIZ);
+
```

Comment:
Please include the null termination here too. It will protect in cases where users pass a longer name than IFNAMSIZ.